spec/integration/parser/environment_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe "A parser environment setting" do
+
+  let(:confdir) { Puppet[:confdir] }
+  let(:environmentpath) { File.expand_path("envdir", confdir) }
+  let(:testingdir) { File.join(environmentpath, "testing") }
+
+  before(:each) do
+    FileUtils.mkdir_p(testingdir)
+  end
+
+  it "selects the given parser when compiling" do
+    manifestsdir = File.expand_path("manifests", confdir)
+    FileUtils.mkdir_p(manifestsdir)
+
```

Comment:
You also want to write out a puppet.conf that has parser set to 'current' ( to the tmp directory)